dotnet/src/Functions/Functions.OpenAPI/Extensions/KernelAIPluginExtensions.cs

```diff
@@ -180,6 +182,17 @@ public static class KernelAIPluginExtensions
     {
         if (TryParseAIPluginForUrl(pluginContents, out var openApiUrl))
         {
+            if (executionParameters?.AuthCallback != null &&
+                TryParseAIPluginForAuth(pluginContents, out var openAIManifestAuth) &&
+                openAIManifestAuth!.Type != OpenAIAuthenticationType.None)
+            {
+                var callback = executionParameters.AuthCallback;
+                executionParameters.AuthCallback = async (request, _) =>
+                {
+                    await callback(request, openAIManifestAuth).ConfigureAwait(false);
+                };
+            }
```

Comment:
Ideally this code shouldn't have any OpenAI specific references.

Can this be implemented something like this:

```
if (executionParameters?.AuthCallback is not null)
{
     var callback = executionParameters.AuthCallback;
     executionParameters.AuthCallback = async (request, _) =>
     {
         await callback(request, pluginContents).ConfigureAwait(false);
     };
}
```

The idea is to push the OpenAI specific code i.e. `TryParseAIPluginForAuth(pluginContents, out var openAIManifestAuth)` into the callback.

Reply:
removing the OpenAI specific references wasn't possible since a single method (i.e. `ImportAIPluginFunctionsAsync`) was being used to load both the OpenAI manifest and the OpenAPI spec, as it required the use of `TryParseAIPluginForUrl`.

due to this, I've separated out the OpenAI specific logic into a `ImportOpenAIPluginFunctionsAsync` function which calls the `ImportOpenApiPluginFunctionsAsync`. @markwallace-microsoft can you take another look at this approach?

Reply:
LGTM